core/src/Kokkos_Core.hpp

```diff
@@ -52,6 +52,10 @@
 
 #include <KokkosCore_Config_DeclareBackend.hpp>
 
+#if defined(KOKKOS_ENABLE_SYCL)
```

Comment:
This is redundant.  By adding the device option, Cmake code in kokkos_tribits.cmake will add Kokkos_Declare_SYCL.hpp to KokkosCore_Config_DeclareBackend.hpp